### PR TITLE
gcc path configurable

### PR DESCRIPTION
**Pull request by jpata**
In some cases on linux, the GCC path is very hard to discover, so the user might want to pre-configure it via environment or `~/.juliarc.jl`.

In particular at CERN, there are 10-s of different versions of gcc that live on a distributed file system with nonstandard naming, making the scanning very slow and brittle.

This alleviates that pain and also preserves the previous discovery mechanism.

### CODE REVIEW

File: src/initialization.jl [hunk 1]
```diff
@@ -266,30 +266,42 @@ function AddLinuxHeaderPaths(C)
...
+                        VersionString = CandidateVersionString
+                        Path = CandidatePath
+                        Triple = triple
+                    end
                 end
             end
         end
     end
-
+    @show VersionString
```
Comment (Keno) [inline src/initialization.jl]: Remove these?

**Comment by jpata** [comment]
Removed the accidentally committed `@show`-s. Seems like the travis failures are related to builds timing out.

**Comment by Keno** [comment]
Yeah, travis failures are a travis problem. This is fine with me, but I wonder if it would be better to store these in `path.jl` to avoid having to set them every time. Thoughts?

**Comment by jpata** [comment]
Do you skipping the scanning altogether and building against a specific version of gcc/stdlib that's always set in `path.jl`?

**Comment by Keno** [comment]
Well, it should still scan during build.jl, but I think it would make sense to detect a GCC version once rather than at every startup.

**Comment by jpata** [comment]
For clusters with nonstandard setup, one would still like to bypass the scan using build-time environment variables.

**Comment by Keno** [comment]
Sure

**Comment by jpata** [comment]
I have some time this weekend, shall I have a go at it (paths in path.jl at build-time) or do you prefer to do it yourself?

**Comment by Keno** [comment]
Please have a go, I'm currently swamped with the remaining 0.5 release work, so I won't get to this for a little bit.

**Comment by Gnimuc** [comment]
closed in favor of #448

**Pull request closed without merging**